chapters/object_types/volumes.txt

```diff
@@ -34,6 +28,7 @@ The scivis volume is created by passing the subtype string `scivis` to
 | opacity.position |`ARRAY1D` of `FLOAT32`|         | optional array to position the elements of `opacity` values in `valueRange`
 | densityScale     |`FLOAT32`             |       1 | makes volumes uniformly thinner or thicker
 | id               |`UINT32`              |     -1u | optional user Id, for <<object_types_frame>> channel `objectId`
+| material         |`MATERIAL`            |         | optional <<object_types_material>> to handle volumetric effects like scattering 
```

Comment:
If this interacts with the material system which implicitly interacts with the sampler concepts, it would make sense to treat the transfer functions as a sampler type. Current samplers can already express the transfer function functionality for the equally spaced case via image1D with a transforms encoding the range and density scale.

While materials (or a material-like volume equivalent) interact with volumes the transfer function parameters seem a bit out of place on the volume object.

The fully symmetric design to the surface API would be to have the volume as a "container" type that just pairs up spatial fields (or an array thereof representing different "attributes") with a material. The material would then encode shader types along the lines of `"integratedDensity"`, `"scatteringMedium"`, `"emissiveMedium"` etc. 

The role of the transfer function would be taken up by samplers which could interact with volume attributes like `attribute0` etc. for the regular channels from the structured field but also `gradient0` etc. for derived quantities.

Reply:
I agree. Your suggestion in my opinion gets us closer to a better design for volumes. Also, having it more symmetric with the surface API makes the spec more intuitive.